kms/mackms/mackms_test.go

```diff
@@ -0,0 +1,705 @@
+//go:build darwin && cgo && !nomackms
+
+// Copyright (c) Smallstep Labs, Inc.
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mackms
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdh"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cf "go.step.sm/crypto/internal/darwin/corefoundation"
+	"go.step.sm/crypto/internal/darwin/security"
+	"go.step.sm/crypto/kms/apiv1"
+	"go.step.sm/crypto/kms/uri"
+	"go.step.sm/crypto/randutil"
+)
+
+func createPrivateKeyOnly(t *testing.T, name string, signatureAlgorithm apiv1.SignatureAlgorithm) (*apiv1.CreateKeyResponse, error) {
+	t.Helper()
+
+	u, err := parseURI(name)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This instance and the others can be replaced with `require.NoError(t, err)` and not return the error.

Reply:
I wasn't sure if the defer functions were called, but I tested, and it does, `t.FailNow()` calls to `runtime.Goexit()`, which runs all deferred calls.